src/Microsoft.DotNet.Watcher.Core/Internal/FileWatcher/PollingFileWatcher.cs

```diff
@@ -117,13 +117,20 @@ private void CheckForChangedFiles()
                 {
                     var fileMeta = _knownEntities[fullFilePath];
 
-                    if (fileMeta.FileInfo.LastWriteTime != f.LastWriteTime)
+                    try
                     {
-                        // File changed
-                        RecordChange(f);
-                    }
+                        if (fileMeta.FileInfo.LastWriteTime != f.LastWriteTime)
+                        {
+                            // File changed
+                            RecordChange(f);
+                        }
 
-                    _knownEntities[fullFilePath] = new FileMeta(fileMeta.FileInfo, true);
+                        _knownEntities[fullFilePath] = new FileMeta(fileMeta.FileInfo, true);
+                    }
+                    catch(FileNotFoundException)
```

Comment:
ok. Should we check for file existence prior to checking `LastWriteTime`. Also formatting (need a space after `catch`)